src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,547 @@
+        dim : int
+            The inferred dimensionality of the problem
+        edgearrays : list of numpy arrays
+            The arrays defining the edge of the grids
+        edgedims   : list of integers, optional, used if eval_type == 'gridsample'
+            The shape of the gridsample grid
+        ngrids : integer, optional, used if eval_type == 'freesample'
+            The number of distinct grids to be constructed
+        gridshape : list of tuple of integers, optional, used if eval_type == 'freesample'
+            The shape of each distinct grid
+        ngridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the 
+        nedgegridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the offset grids
+        x0 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the first vertex of the grids
+        x1 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the last vertex of the grids
+
+        
+        """
+
+        # check the validity of the input
+        if len(cells)==0:
+            raise ValueError("LintSampler._setgrid: you must specify an evaluation domain with a single array, a tuple of arrays or a list of tuples of arrays. See documentation for details.")
+            # or, we could here drop into an adaptive grid selection
+            # or, we could sample on a unit hypercube with dimensions of the pdf
+
+        # 0. are we in the 1d case? the input is just a single tuple or 1d array.
+        # i.e. cells = np.linspace(-12,12,100)
+        if hasattr(cells, "__len__") and not hasattr(cells[0], "__len__"):
+            self.eval_type='gridsample'
+
+            # override the inferred dimensionality
+            self.dim = 1
+
+            # set the arrays
+            # we are being passed the array directly
+            self.edgearrays = np.array(cells)
+            self.edgedims = len(cells)
+
+
+        # 1. tuples defining the array -> make arrays, pass to gridsample
+        # i.e. cells = (np.linspace(-12,12,100),np.linspace(-4,4,50))
+        elif isinstance(cells,tuple):
+
+            if not (hasattr(cells[0], "__len__") and not hasattr(cells[0][0], "__len__")):
+                raise TypeError("LintSampler: Cells must be a single tuple of 1D sequences (arrays, lists, or tuples).")
+            
+            # infer the dimensionality
+            self.dim = len(cells)
+
+            # set the evaluation type
+            self.eval_type = 'gridsample'
+
+            # keep track of the edge arrays
+            self.edgearrays = []
+
+            # keep track of the dimensions
+            self.edgedims = np.ones(self.dim,dtype='int')
+
+            # create the grid
+            for d in range(0,self.dim):
+                self.edgearrays.append(cells[d])
```

Comment:
In case user passes tuple of 1D lists or tuple of 1D tuples (instead of tuple of arrays)
```suggestion
                self.edgearrays.append(np.array(cells[d]))
```